### PR TITLE
set_vote_parameters_validates_input_52

### PR DESCRIPTION
**Pull request by valentinbujdoso**

### CODE REVIEW

**Comment by magwas** [comment]
You have a mutant alive.

**Comment by magwas** [comment]
this is #52

File: src/test/java/org/rulez/demokracia/pdengine/VoteSetParametersTest.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+						canVote,
+						canView)
+			).assertMessageIs(String.format("Illegal adminKey: %s", invalidAdminKey));
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("set vote parameters")
+	@tested_behaviour("vote invariants")
+	@Test
+	public void setVoteParameters_sets_the_minEndorsement_parameter_of_the_vote() {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteSetParametersTest.java:110]: 1. This is not the "vote invariants" behaviour. The most close behaviour is "update vote title" https://adadocs.demokracia.rulez.org/PDEngine/edemo/master/index.html#04c661ac-e467-430b-a514-9b8e6c84de5f
That behaviour directly contradicts to the tested behaviour.
However this time the model is wrong. I am modeling a "updates vote parameters" behaviour shortly. Please use that.

File: src/test/java/org/rulez/demokracia/pdengine/VoteSetParametersTest.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+		assertEquals(minEndorsements, vote.minEndorsements);
+	}
+
+	@tested_feature("Manage votes")
+	@tested_operation("set vote parameters")
+	@tested_behaviour("vote invariants")
+	@Test
+	public void setVoteParameters_sets_the_canAddIn_parameter_of_the_vote() {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteSetParametersTest.java:118]: same as above

File: src/test/java/org/rulez/demokracia/pdengine/VoteSetParametersTest.java [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+		assertEquals(true, vote.canAddin);
+	}
+
+	@tested_feature("Manage votes")
+	@tested_operation("set vote parameters")
+	@tested_behaviour("vote invariants")
+	@Test
+	public void setVoteParameters_sets_the_canEndorse_parameter_of_the_vote() {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteSetParametersTest.java:126]: same as above

File: src/test/java/org/rulez/demokracia/pdengine/VoteSetParametersTest.java [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+		assertEquals(true, vote.canEndorse);
+	}
+
+	@tested_feature("Manage votes")
+	@tested_operation("set vote parameters")
+	@tested_behaviour("vote invariants")
+	@Test
+	public void setVoteParameters_sets_the_canVote_parameter_of_the_vote() {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteSetParametersTest.java:134]: same as above

File: src/test/java/org/rulez/demokracia/pdengine/VoteSetParametersTest.java [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+		assertEquals(true, vote.canVote);
+	}
+
+	@tested_feature("Manage votes")
+	@tested_operation("set vote parameters")
+	@tested_behaviour("vote invariants")
+	@Test
+	public void setVoteParameters_sets_the_canView_parameter_of_the_vote() {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteSetParametersTest.java:142]: same as above

**Review by magwas — DISMISSED** [review]
sorry about the modeling error

File: src/test/java/org/rulez/demokracia/pdengine/VoteSetParametersTest.java (hunk 1 above)
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteSetParametersTest.java:110]: This is "sets the parameters of the vote" behaviour.
https://adadocs.demokracia.rulez.org/PDEngine/edemo/master/index.html#f51eabe8-c035-439b-b09a-3c3130c18a3e

**Review by magwas — CHANGES_REQUESTED** [review]
Please correct the marked behaviours.

File: src/test/java/org/rulez/demokracia/pdengine/VoteSetParametersTest.java (hunk 1 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteSetParametersTest.java:110]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/VoteSetParametersTest.java (hunk 5 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteSetParametersTest.java:142]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/VoteSetParametersTest.java (hunk 4 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteSetParametersTest.java:134]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/VoteSetParametersTest.java (hunk 3 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteSetParametersTest.java:126]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/VoteSetParametersTest.java (hunk 2 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteSetParametersTest.java:118]: Fixed

**Comment by valentinbujdoso** [comment]
@magwas Could you help me how i can solve the SonarCloud problem?
I know if I delete the minEndorsements check from VoteRegistry.java, but I think we need it there.

**Review by magwas — APPROVED** [review]
good work

**Comment by magwas** [comment]
testcaseCount=14

**Pull request merged**